src/resources/js/download.js

```diff
@@ -27,6 +28,30 @@ $.get("/api/packages").done(function(json) {
 			
 			var $optpkg = $(optpkgTemplate);
 			$('.optpkg-name', $optpkg).text(pkg.path);
+			if (preselectedPackage.includes(pkg.path)) {
+				$('.optpkg-check', $optpkg).prop("checked", true);
+				$('.optpkg-check', $optpkg).closest('.optpkg').toggleClass("selected");
+			}
+
+			$('.optpkg-check', $optpkg).change({pkg: pkg}, (event) => {
+				const element = $(event.currentTarget);
+				let newUrl = new URL(window.location.href);
+				let currentSelected = newUrl.searchParams.getAll("package") ;
+				newUrl.searchParams.delete("package");
+				const pkgPath = event.data.pkg.path;
+				if (element.is(':checked')) {
+					if (!currentSelected.includes(pkgPath)) {
+						currentSelected = [...currentSelected, pkgPath];
```

Comment:
FYI https://caniuse.com/?search=spread%20operator

Not sure what your ideal target for browser support is @mholt but just FYI

Reply:
Good find, thanks. That should be okay though; it's also a fairly niche feature. Browsers should be auto-updating these days anyway.